File: PHP/program/Crud.php
<?php
// import file
require ('Mahasiswa.php');

// deklarasi kelas crud
class Crud {
     // atribute private
    private $arraymhs = array(); // list of objek mahasiswa

    /* konstruktor */
    public function __construct( ) {
    }

    // method create : membuat objek mahasiswa
    public function create($nim, $nama, $prodi, $fakultas, $foto){
        // instansiasi objek mahasiswa
        $mhs = new Mahasiswa($nim, $nama, $prodi, $fakultas, $foto);
         // masukan objek ke dalam list
        array_push($this->arraymhs, $mhs);
    }

    // method read : menampilkan isi list
    public function read(){
        // menampilkan isi list of objek mahasiswa dalam tabel
        echo "<center><table border='3'>";
        echo "<tr>
            <th> NIM </th>
            <th> Nama </th>
            <th> Prodi </th>
            <th> Fakultas </th>
            <th> Foto </th>
            </tr>";
            for ($i=0; $i < count($this->arraymhs); $i++) { 
                echo "<tr><td>";
                echo $this->arraymhs[$i]->get_nim();
                echo "</td><td>";
                echo $this->arraymhs[$i]->get_nama();
                echo "</td><td>";
                echo $this->arraymhs[$i]->get_prodi();
                echo "</td><td>";
                echo $this->arraymhs[$i]->get_fakultas();
                echo "</td><td>";
                echo "<img src='" . $this->arraymhs[$i]->get_foto(). "'style='width: 150px'>";
                echo "</td>";
                echo "</tr>";
            }
        echo "</table> </center>";
        }

    // method update : update objek yang ingin diubah
    public function update($nim, $nama, $prodi, $fakultas, $foto ){
        // perulangan untuk mencari objek yang akan di ubah berdasarkan nim yang diinput
        for ($i=0; $i < count($this->arraymhs); $i++) { 
            // apabila nim pada objek nya sama dengan nim inputan
            if ($this->arraymhs[$i]->get_nim() == $nim) {
                // update data yang ingin diubah dengan menggunakan method setter di kelas mahasiswa
                $this->arraymhs[$i]->set_nama($nama);
                $this->arraymhs[$i]->set_prodi($prodi);
                $this->arraymhs[$i]->set_fakultas($fakultas);
                $this->arraymhs[$i]->set_foto($foto);
                break;
            }
        }
    }

    // method delete : delete objek yang ingin dihapus
    public function delete($nim){
        // perulangan untuk mencari objek yang akan di hapus berdasarkan nim yang diinput
        for ($i=0; $i < count($this->arraymhs); $i++) { 
            // apabila nim pada objek nya sama dengan nim inputan
            if ($this->arraymhs[$i]->get_nim() == $nim) {
                array_splice($this->arraymhs, $i,1); // hapus objek pada list
              break;
            }
        }
    }
}

?>

File: CPP/program/Crud.cpp
// import library dan file
#include <iostream>
#include <string>
#include "Mahasiswa.cpp"

// menggunakan standarad namespace
using namespace std;

// deklarasi kelas crud
class Crud
{
    // atribute private
private:
    list<Mahasiswa> llist; // list of objek mahasiswa

public:
    /* konstruktor tanpa parameter */
    Crud()
    {
    }

    // method create : membuat objek mahasiswa
    void create(string nim, string nama, string prodi, string fakultas)
    {
        // instansiasi objek mahasiswa, lalu masukan objek ke dalam list
        llist.push_back(Mahasiswa(nim, nama, prodi, fakultas));
    }

    // method read : menampilkan isi list
    void read()
    {
        // menampilkan isi list of objek mahasiswa
        cout << "\nList Mahasiswa : " << '\n';
        int i = 0;
        for (list<Mahasiswa>::iterator it = llist.begin(); it != llist.end(); it++)
        {
            cout << (i + 1) << ".| " << it->get_nim() << " | " << it->get_nama() << " | " << it->get_prodi() << " | " << it->get_fakultas() << '\n';
            i++;
        }
    }

    // method update : update objek yang ingin diubah
    void update(string nim)
    {
        // deklrasi variabel untuk menampung nama, prodi, dan fakultas dari data yang akan diubah
        string nama, prodi, fakultas;

        // perulangan untuk mencari objek yang akan  diubah berdasarkan nim yang diinput
        for (list<Mahasiswa>::iterator it = llist.begin(); it != llist.end(); it++)
        {
            // apabila nim pada objek nya sama dengan nim inputan
            if (nim == it->get_nim())
            {
                // input isi variabel nama prodi fakultas
                cout << "\nMasukan data yang ingin diubah !\n";

                cout << "Nama: ";
                cin >> nama;
                cout << "Prodi: ";
                cin >> prodi;
                cout << "Fakultas: ";
                cin >> fakultas;

                // update data objek yang ingin diubah dengan menggunakan method setter di kelas
                // mahasiswa
                it->set_nama(nama);
                it->set_prodi(prodi);
                it->set_fakultas(fakultas);
                break;
            }
        }
    }
    // method delete : delete objek yang ingin dihapus
    void del(string nim)
    {
        // perulangan untuk  mencari objek yang akan di hapus berdasarkan nim yang diinput
        for (list<Mahasiswa>::iterator it = llist.begin(); it != llist.end(); it++)
        {
            // apabila nim pada objek nya sama dengan nim inputan
            if (nim == it->get_nim())
            {
                llist.erase(it); // hapus objek pada list
                break;
            }
        }
    }

    ~Crud()
    {
    }
};
